### PR TITLE
LPS-140211 Prevent publish an Object without any field

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-140211

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:sf

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -408,6 +409,14 @@ public ObjectDefinition publishCustomObjectDefinition(
...
 		}
 
+		int objectFieldsAmount =
+			_objectFieldPersistence.countByObjectDefinitionId(
+				objectDefinition.getObjectDefinitionId());
+
+		if (objectFieldsAmount == 0) {
+			throw new RequiredObjectFieldException();
+		}
+
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:419]: @p-albuquerque can you suppress this variable and put it inside the if condition? Like this:

```
if (_objectFieldPersistence.countByObjectDefinitionId(
            objectDefinition.getObjectDefinitionId()) == 0) {
			throw new RequiredObjectFieldException();
		}
```

**Review by gabrielwas — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java (hunk 1 above)
Comment (p-albuquerque) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:419]: I tried this, however the FormatSource failed to build, with the follow formatting issue:
"Create a new variable for the left hand side operand of the '==' operator for better readability"

Comment (p-albuquerque) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:419]: I've tried this, however the FormatSource failed to build, with the follow formatting issue:
"Create a new variable for the left hand side operand of the '==' operator for better readability"

**Comment by p-albuquerque** [comment]
ci:test:relevant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java (hunk 1 above)
Comment (gabrielwas) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:419]: Ok! Thx, Paulo!

Comment (gabrielwas) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:419]: Can you rename it for just `count` ?

**Comment by gabrielwas** [comment]
ci:stop

**Comment by p-albuquerque** [comment]
ci:test:sf

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java (hunk 1 above)
Comment (p-albuquerque) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java]: Done.

**Comment by gabrielwas** [comment]
LGTM :)

**Comment by gabrielwas** [comment]
ci:test:relevant

**Review by gabrielwas — APPROVED** [review]

**Comment by marcelabc** [comment]
I'm removing the flag 'ready to forward' based on Gabriel's comment.

cc: @p-albuquerque @gabrielwas

**Comment by gabrielwas** [comment]
@marcelabc thx!

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by gabrielwas** [comment]
@p-albuquerque can you check these Poshio failures?

**Comment by p-albuquerque** [comment]
Waiting for QA answer/guidance.

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
Hey @p-albuquerque can you check this test failure?
There is only one test failing and doesn't seem like a flaky test since I ran `ci:test:relevant` again and the same test has failed.

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by gabrielwas** [comment]
ci:forward

**Pull request closed without merging**